p2p/src/net/default_backend/backend.rs

```diff
@@ -115,6 +115,9 @@ pub struct Backend<T: TransportSocket> {
     /// TX channel for sending syncing events
     sync_tx: mpsc::UnboundedSender<SyncingEvent>,
 
+    /// TX channel for broadcasting p2p events
+    p2p_tx: broadcast::Sender<P2pEvent>,
```

Comment:
I'm not really a big fan of this. This is because P2pEvent is exclusively meant for external subsystems. P2pEvents have no order requirements for components within p2p, for example, and hence can be delayed. Despite this being just an enum, it's made for P2pEventHandler, not to be thrown around. This is a leak in the opposite direction (to the inside, instead of to the outside).

If you need this information for other p2p components (e.g., clients connected/disconnected), then perhaps you should create a separate abstraction for that, which would be relevant for individual components within p2p, not as a general P2pEvent abstraction for both inside and outside.